lib/phoenix/digester.ex

```diff
@@ -148,18 +151,24 @@ defmodule Phoenix.Digester do
     path = Path.join(output_path, file.relative_path)
     File.mkdir_p!(path)
 
-    # compressed files
-    if compress_file?(file) do
-      File.write!(
-        Path.join(path, file.digested_filename <> ".gz"),
-        :zlib.gzip(file.digested_content)
-      )
-
-      File.write!(
-        Path.join(path, file.filename <> ".gz"),
-        :zlib.gzip(file.content)
-      )
-    end
+    compressors = Application.fetch_env!(:phoenix, :static_compressors)
+
+    Enum.each(compressors, fn(compressor) ->
+      if compressor.compress_file?(file.absolute_path, file.content, file.digested_content) do
```

Comment:
Do we need to pass content and digested_content? :)

Reply:
This is something I was less sure about. I took some inspiration from [cowboy's](https://github.com/ninenines/cowboy/blob/master/src/cowboy_compress_h.erl#L119-L126) (non-static) response compression and [Webpack compression plugin](https://webpack.js.org/plugins/compression-webpack-plugin/), where there is a threshold for the size. The webpack plugin also allows specifying a minimum compression ratio.

A quick example is robots.txt, where if you want no crawling, the content is:
```
User-agent: *
Disallow: /
```

Which ends up being 26 bytes, and running `mix phx.digest` generates robots.txt.gz which ends up being larger at 46 bytes.

In terms of supplying both content and digested_content, I was erring on supplying more information, as they potentially are different sizes or end up compressing slightly differently. I also considered passing the whole `file` map, but I was worried about exposing too much of the structure publicly and potentially locking things in.

Reply:
@mitchellhenke really smart! Maybe a better API is then to call `compress_file(filename, contents)` which returns `{:ok, result} | :error`? For example, could I want to encode one but not the other?